classes/01class/exercises/c01-aws02/zarajoy/README.md

```diff
@@ -0,0 +1,164 @@
+# C01-AWS02
+
+## Commands Execution Output
+
+- Commands to create the S3 Bucket and to copy the file to the bucket:
+
+https://awscli.amazonaws.com/v2/documentation/api/latest/reference/s3api/create-bucket.html
+```
+z@bacon:~$ aws s3api create-bucket --bucket zarajoy-aws02-bucket --region ap-southeast-2 --create-bucket-configuration LocationConstraint=ap-southeast-2
+{
+    "Location": "http://zarajoy-aws02-bucket.s3.amazonaws.com/"
+}
+z@bacon:~$ aws s3 cp ./Documents/upload.txt s3://zarajoy-aws02-bucket
+
+upload: Documents/upload.txt to s3://zarajoy-aws02-bucket/upload.txt
+
+z@bacon:~$ aws s3 ls zarajoy-aws02-bucket
+
+2020-06-19 12:53:14         16 upload.txt
+```
+
+- Commands to allow the EC2 instance to access the files in S3:
+
+https://aws.amazon.com/premiumsupport/knowledge-center/ec2-instance-access-s3-bucket/
+
+https://aws.amazon.com/premiumsupport/knowledge-center/iam-assume-role-cli/
+
+https://docs.aws.amazon.com/cli/latest/reference/iam/create-role.html
+
+https://aws.amazon.com/blogs/security/organize-your-permissions-by-using-separate-managed-policies/
+
+```
+//create the role-policy-document.json file
+z@bacon:~$ vim assume_role_policy.json
+{
+  "Version": "2012-10-17",
+  "Statement": [
+    {
+      "Sid": "",
+      "Effect": "Allow",
+      "Principal": {
+        "Service": "ec2.amazonaws.com"
+      },
+      "Action": "sts:AssumeRole"
+    }
+  ]
+}
+// to save and exit esc :w +enter :x+enter
+
+//create IAM role and link policy document created above
+z@bacon:~$ aws iam create-role --role-name access-zarajoy-aws02-bucket --assume-role-policy-document file://assume_role_policy.json
+{
+    "Role": {
+        "Path": "/",
+        "RoleName": "access-zarajoy-aws02-bucket",
+        "RoleId": "AROAXOYORRTGGTDS34HA4",
+        "Arn": "arn:aws:iam::512742231244:role/access-zarajoy-aws02-bucket",
+        "CreateDate": "2020-06-19T04:26:36Z",
+        "AssumeRolePolicyDocument": {
+            "Version": "2012-10-17",
+            "Statement": [
+                {
+                    "Sid": "",
+                    "Effect": "Allow",
+                    "Principal": {
+                        "Service": "ec2.amazonaws.com"
+                    },
+                    "Action": "sts:AssumeRole"
+                }
+            ]
+        }
+    }
+}
+
+z@bacon:~$ aws iam attach-role-policy --policy-arn arn:aws:iam::aws:policy/AmazonS3ReadOnlyAccess --role-name access-zarajoy-aws02-bucket
```

Comment:
you can always use existing policies, but you normally will need to create custom policies to be able to achieve a least privilege model.
Can you, instead of use an existent policy, create a new one with only the permissions you need to the resources you need?
let me know if not clear.. :)

Reply:
i have done this now, making my own policy with get and list object for my bucket. i redid the whole exercise (practise makes perfect?)